### PR TITLE
Push notifications: iOS request permissions, surface token and flags

### PR DESCRIPTION
**Pull request by oskarth**
### Summary
Push notifications: iOS request permissions, surface token and flags

- Asks for permissions on iOS at startup
- Introduce NOTIFICATIONS_WIP_ENABLED flag
- Surface FCM Token and put it in clipboard for QA manual end to end testing

### Note to reviewers

This is a WIP and part of spiking out push notifications end to end. The goal of this is to let QA do end to end testing on devices and surface problems now rather than later. The next step is to integrate with status-go code to trigger notifications on messages from that side, as it appears more desirable than doing it from RN side (via RN Firebase Cloud Functions). This will be addressed in future issues and PRs.

### Steps to test for QA
1. Load app and login, you should see a popup with a token. If on iOS it'll ask you for permissions, grant it.

2. The token is in your clipboard, paste it so it ends up at your computer.

3. At your computer, go to https://console.firebase.google.com/u/1/ under Status account, should see Status React App. If you have issues with this please message me.

4. Background the app.

5. Grow -> Notifications -> New message.

6. Enter message text, target single device and paste in FCM registration token and send message.

7. You should see a notification and message completed at Firebase.

Note: iOS works fine, Android seems to depend on Google Play Service. I suspect there are some issues there so logs would be appreciated. I don't think we'll be able to get it to work for all Android devices/setups with just Firebase, though. So consider this a work in progress.

status: ready

### CODE REVIEW

**Review by jeluard — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Review by yenda — APPROVED** [review]

**Comment by oskarth** [comment]
Note: don't merge. Accidentally pushed cherry-picked commits that uses latest status-go for testing - will remove these when squashing.

Same with log level (but on purpose to test Testfairy).

File: src/status_im/ui/screens/events.cljs [hunk 1]
```diff
@@ -271,6 +272,8 @@
 (register-handler-fx
  :get-fcm-token
  (fn [_ _]
+   ;; XXX: compiled away?
+   (utils/show-popup "TODO" "FCM FX CALLED")
```
Comment (oskarth) [inline src/status_im/ui/screens/events.cljs]: QUESTION: Expect this to be called but it isn't on release build. The popups in `initialize-account` and `init-wallet` popups are called. Running in debug mode on real device shows this popup. Why?

It is part of the compiled bundle:
```
Oskars-MacBook-Pro:status-react oskarth$ less status.xcarchive/Products/Applications/StatusIm.app/main.jsbundle | tr ',' '\n' | grep "FCM FX" -C 3
status_im.utils.handlers.register_handler_fx.cljs$core$IFn$_invoke$arity$2(cljs.core.cst$kw$get_DASH_fcm_DASH_token
function(e
c){return status_im.utils.utils.show_popup("TODO"
"FCM FX CALLED")
status_im.utils.notifications.get_fcm_token()
cljs.core.PersistentArrayMap.EMPTY})
status_im.components.status_bar={}
```

Comment (oskarth) [reply src/status_im/ui/screens/events.cljs]: One can see that they should all be called in succession, which they are in debug mode but not for release for whatever reason: https://github.com/status-im/status-react/blob/feature/push-notifications-spike-2-%231599/src/status_im/ui/screens/events.cljs#L168-L170

Comment (oskarth) [reply src/status_im/ui/screens/events.cljs]: Seems like the reason was too many alerts in quick succession. iOS appears to have a max of two alerts up simultaneously, and which ones depends on delay between them.

**Comment by oskarth** [comment]
Squashed and removed latest status-go / log commits. Ready to merge as soon as QA has tested it on iOS at least.

**Comment by asemiankevich** [comment]
This is tested and passed. I was able to receive notification for both iOS and Android using mentioned workflow through Firebase

**Pull request merged**